bgpd/bgp_aspath.c

```diff
@@ -1558,6 +1561,38 @@ struct aspath *aspath_prepend(struct aspath *as1, struct aspath *as2)
 	/* Not reached */
 }
 
+/* insert aspath exclude in head of orphan exclude list*/
+void as_exclude_set_orphan(struct aspath_exclude *ase)
+{
+	ase->exclude_aspath_acl = NULL;
+	as_list_list_add_head(&as_exclude_list_orphan, ase);
+}
+
+void as_exclude_remove_orphan(struct aspath_exclude *ase)
+{
+	if (as_list_list_count(&as_exclude_list_orphan))
```

Comment:
Why is it necessary to have this test?  Why not just delete it?

Reply:
from Francois:
orphan list may be destroyed before the as_path_exclude that we are destroying currently.
 this occurs at termination time.
 this test verify the existence of the list .